lib/solana.cpp

```diff
@@ -180,8 +343,23 @@ json Connection::simulateTransaction(
   // create request
   const json params = {b64Tx, config.toJson()};
   const auto reqJson = jsonRequest("simulateTransaction", params);
+  const json res = sendJsonRpcRequest(reqJson)["value"];
+  std::string err = "";
+  std::vector<std::string> accounts;
+  std::vector<std::string> logs;
+  if (!res["err"].is_null()) {
+    err = res["err"];
+  }
+
+  if (!res["accounts"].is_null()) {
+    accounts = res["accounts"].get<std::vector<std::string>>();
+  }
+  if (!res["logs"].is_null()) {
+    logs = res["logs"].get<std::vector<std::string>>();
+  }
+  int unitsconsumed = res["unitsConsumed"];
```

Comment:
always use explicit byte width types to avoid compiler differences, int is only guaranteed to hold 16 bits, can easily overflow here